android-static-binding-generator/project/parser/visitors/es5-visitors.js

```diff
@@ -171,23 +178,69 @@ var es5_visitors = (function () {
         }
     }
 
-	function traverseForDecorate(path, config, depth) {
-		var iifeRoot = _getParrent(path, depth)
-		var body = iifeRoot.node.body;
-		for (var index in body) {
-			var ci = body[index];
-			if (isDecorateStatement(ci)) {
-				// returns the node of the decorate (node.expression.right.callee)
-				// __decorate([..])
-				return ci.expression.right.arguments[0].elements;
-			}
-		}
-
-		return null;
-	}
+    function getTypeScriptExtendSuperCallLocation(extendPath, config) {
+        var constructorFunctionName;
+        var returnIdentifierName;
+        var superCalleeStartColumn;
+        var superCalleeLine;
+
+        // checks if constructor function and return identifiers match in a transpiled typescript class extend
+        if (extendPath.node.body && extendPath.node.body.length >= 3) {
+            if (types.isFunctionDeclaration(extendPath.node.body[1]) && extendPath.node.body[1].id)
+                constructorFunctionName = extendPath.node.body[1].id.name;
+            if (types.isReturnStatement(extendPath.node.body[extendPath.node.body.length - 1]))
+                returnIdentifierName = extendPath.node.body[extendPath.node.body.length - 1].argument.name;
+        }
+
+        // checks the typescript-extended class for a strict format, and a super call/apply inside
+        /**
+         *      function MyBtn() {
+                    var _this = _super.call(this) || this;
+                    return global.__native(_this);
+                }
+         */
+        if (constructorFunctionName === returnIdentifierName && !!constructorFunctionName) {
+            var constructorFunctionScope = extendPath.node.body[1];
+            var firstLineOfCtorFunction = constructorFunctionScope.body.body[0];
+            if (types.isVariableDeclaration(firstLineOfCtorFunction)) {
+                var variableRHS = firstLineOfCtorFunction.declarations[0].init;
+                var variableRHSPaths = variableRHS._paths.filter(node => types.isCallExpression(node));
+                variableRHSPaths = variableRHSPaths.length > 0 ? variableRHSPaths[0] : null;
+                if (types.isMemberExpression(variableRHSPaths && variableRHSPaths.node.callee)) {
```

Comment:
``` TypeScript
class X extends Y {
    constructor(a: () => number) {
        var x = a.call(null);
        super(x + 2);
    }
}
```

Generates:
```
var X = /** @class */ (function (_super) {
    __extends(X, _super);
    function X(a) {
        var _this = this;
        var x = a.call(null);
        _this = _super.call(this, x + 2) || this;
        return _this;
    }
    return X;
}(Y));
```
I am not sure that the `var x = a.call(null);` won't be considered as the super call in the current implementation. Perhaps we should add some check for the "_this" identifier on the left hand side of the assignment and probably the "_super" identifier on the left hand side of the member access.